docs/public/components/using-components.md

```diff
@@ -0,0 +1,170 @@
+# Using components
+
+Using components within other components is covered on the [create components page](creating-components.md).
+
+Now let's explore how to use components in your binaries. One of the core idea behind component design is to be able to
+create new binaries for customers by aggregating components.
+
+## the `cmd` folder
+
+All `main` functions and binary entry points should be in the `cmd` folder.
+
+The `cmd` folder uses the following hierarchy:
+
+```
+cmd /
+    <binary name> /
+        main.go                   <-- The entry points from your binary
+        subcommands /             <-- All subcommand for your binary CLI
+            <subcommand name> /   <-- The code specific to a single subcommand
+                command.go
+                command_test.go
+```
+
+Say you want to add a `test` command to the `agent` CLI.
+
+You would create the following file:
+
+=== ":octicons-file-code-16: cmd/agent/subcommands/test/command.go"
+    ```go
+    package test
+
+    import (
+    // [...]
+    )
+
+    // Commands returns a slice of subcommands for the 'agent' command.
+    //
+    // The Agent uses "cobra" to create its CLI. The command method is your entrypoint. Here, you're going to create a single
+    // command.
+    func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+        cmd := &cobra.Command{
+            Use:   "test",
+            Short: "a test command for the Agent",
+            Long:  ``,
+            RunE: func(_ *cobra.Command, _ []string) error {
+                return fxutil.OneShot(
+                    <callback>,
+                    <list of dependencies>.
+                )
+            },
+        }
+
+        return []*cobra.Command{cmd}
+    }
+    ```
+
+The code above creates a test command that does nothing. As you can see, `fxutil.OneShot` helpers are being used. These
+helpers initialize an Fx app with all the wanted dependencies. 
+
+The next section explains how to request a
+dependency.
+
+## Importing components
+
+The `fxutil.OneShot` takes a list of components and gives them to Fx. Note that this only tells Fx how to create types when they're needed. This does not do anything else.
+
+In order for a components to be instantiated it needs to be required:
+
++ Required as a parameter by the `callback` function
```

Comment:
You may give `fx.Supply` as an example because `callback` is a very general term.